hail/src/main/scala/is/hail/expr/ir/Compile.scala

```diff
@@ -177,18 +181,24 @@ object CompileIterator {
   ): (PType, (Int, Region) => F) = {
 
     val fb = EmitFunctionBuilder.apply[F](ctx, "stream", argTypeInfo.toFastIndexedSeq, CodeParamType(BooleanInfo))
+    val outerRegionField = fb.genFieldThisRef[Region]("outerRegion")
+    val eltRegionField = fb.genFieldThisRef[Region]("eltRegion")
+    val setF = fb.newEmitMethod("setRegions", FastIndexedSeq(CodeParamType(typeInfo[Region]), CodeParamType(typeInfo[Region])), CodeParamType(typeInfo[Unit]))
+    setF.emit(Code(outerRegionField := setF.getCodeParam[Region](1), eltRegionField := setF.getCodeParam[Region](2)))
+
     val stepF = fb.apply_method
     val stepFECB = stepF.ecb
 
-    val region = StagedRegion(EmitRegion.default(stepF).region, allowSubregions = false)
+    val eltRegion = StagedRegion(eltRegionField, allowSubregions = false)
```

Comment:
we'll flip these to `true` when we solve the mixed memory management design, right?

Reply:
I don't know yet what the final design will look like for setting and passing down the allocation strategy, but as things are now I think we'd want `outerRegion` to allow subregions, but not `eltRegion`.